chemberta/finetune/finetune.py

```diff
@@ -0,0 +1,171 @@
+
+FLAGS = flags.FLAGS
+
+# Settings
+flags.DEFINE_string(name="output_dir", default="default_dir", help="")
+flags.DEFINE_boolean(name="overwrite_output_dir", default=True, help="")
+flags.DEFINE_string(name="run_name", default="default_run", help="")
+flags.DEFINE_integer(name="seed", default=0, help="")
+
+# Model params
+flags.DEFINE_string(name="model_dir", default=None, help="")
+flags.DEFINE_boolean(name="freeze_base_model", default=False, help="")
+
+# Train params
+flags.DEFINE_integer(name="logging_steps", default=10, help="")
+flags.DEFINE_integer(name="num_train_epochs", default=3, help="")
+flags.DEFINE_integer(name="per_device_train_batch_size", default=64, help="")
+flags.DEFINE_integer(name="per_device_eval_batch_size", default=64, help="")
+
+# Dataset params
+flags.DEFINE_string(name="dataset", default=None, help="")
+flags.DEFINE_string(name="split", default="scaffold", help="")
+
+# Tokenizer params
+flags.DEFINE_string(
+    name="tokenizer_path",
+    default="seyonec/SMILES_tokenized_PubChem_shard00_160k",
+    help="",
+)
+flags.DEFINE_integer(name="max_tokenizer_len", default=512, help="")
+
+flags.mark_flag_as_required("dataset")
+flags.mark_flag_as_required("model_dir")
+
+
+def main(argv):
+    torch.manual_seed(FLAGS.seed)
+    run_dir = os.path.join(FLAGS.output_dir, FLAGS.run_name)
+    
+    tasks, (train_df, valid_df, test_df), transformers = load_molnet_dataset(FLAGS.dataset, split=FLAGS.split, df_format="chemprop")
+    assert(len(tasks) == 1)
+    
+    tokenizer = RobertaTokenizerFast.from_pretrained('seyonec/SMILES_tokenized_PubChem_shard00_160k', max_len=FLAGS.max_tokenizer_len)
+    
+    train_encodings = tokenizer(train_df["smiles"].tolist(), truncation=True, padding=True)
+    valid_encodings = tokenizer(valid_df["smiles"].tolist(), truncation=True, padding=True)
+    test_encodings = tokenizer(test_df["smiles"].tolist(), truncation=True, padding=True)
+    
+    train_labels = train_df.iloc[:, 1].values
+    valid_labels = valid_df.iloc[:, 1].values
+    test_labels = test_df.iloc[:, 1].values
+    
+    train_dataset = MolNetDataset(train_encodings, train_labels)
+    valid_dataset = MolNetDataset(valid_encodings)
+    test_dataset = MolNetDataset(test_encodings)
+    
+    config = RobertaConfig.from_pretrained(FLAGS.model_dir)
+    
+    dataset_type = get_dataset_info(FLAGS.dataset)["dataset_type"]
+    if dataset_type == "classification":
+        model_class = RobertaForSequenceClassification
+    elif dataset_type == "regression":
+        model_class = RobertaForRegression
+        config.num_labels = 1
+        config.norm_mean = [np.mean(np.array(train_labels), axis=0)]
+        config.norm_std = [np.std(np.array(train_labels), axis=0)]
+
+    model = model_class.from_pretrained(FLAGS.model_dir, config=config)
+    
+    if FLAGS.freeze_base_model:
+        for name, param in model.base_model.named_parameters():
+            param.requires_grad = False
```

Comment:
i'm 90% sure this isn't a problem - but I just wanted to make sure that we're positive that the new parameters that get added are trainable

Reply:
yes - I made sure to check this. `base_model` includes only the pretrained "base" model!

Reply:
yup! this is the config structure for HF models - base_model is just the pre-trained weights